Block/CategoryMapping.php

```diff
@@ -0,0 +1,136 @@
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON
+ * ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ * @author Nosto Solutions Ltd <contact@nosto.com>
+ * @copyright 2019 Nosto Solutions Ltd
+ * @license http://opensource.org/licenses/BSD-3-Clause BSD 3-Clause
+ *
+ */
+
+namespace Nosto\Cmp\Block;
+
+use Magento\Catalog\Model\ResourceModel\Category\CollectionFactory;
+use Magento\Framework\Exception\NoSuchEntityException;
+use Magento\Framework\View\Element\Template;
+use Magento\Framework\View\Element\Template\Context;
+use Magento\Store\Model\Store;
+use Magento\Store\Model\StoreManagerInterface;
+use Nosto\Tagging\Logger\Logger as NostoLogger;
+use Nosto\Tagging\Model\Service\Product\Category\DefaultCategoryService as CategoryBuilder;
+use Magento\Catalog\Model\Category;
+use Magento\Framework\Exception\LocalizedException;
+
+class CategoryMapping extends Template
+{
+    /** @var StoreManagerInterface */
+    private $storeManager;
+
+    /** @var CollectionFactory */
+    private $collectionFactory;
+
+    /** @var CategoryBuilder */
+    private $categoryBuilder;
+
+    /** @var NostoLogger */
+    private $logger;
+
+    /**
+     * CategoryMapping constructor.
+     */
+    public function __construct(
+        StoreManagerInterface $storeManager,
+        CollectionFactory $collectionFactory,
+        CategoryBuilder $categoryBuilder,
+        Context $context,
+        NostoLogger $logger
+    ) {
+        parent::__construct($context);
+        $this->storeManager = $storeManager;
+        $this->collectionFactory = $collectionFactory;
+        $this->categoryBuilder = $categoryBuilder;
+        $this->logger = $logger;
+    }
+
+    /**
+     * @return false|string
+     */
+    public function getCategoryMap() {
+
+        $array = [];
+        try {
+            $store = $this->storeManager->getStore();
+            if ($store instanceof Store) {
+                $array = $this->getMagentoCategories($store);
+            }
+        } catch (NoSuchEntityException $e) {
+            $this->logger->exception('Could not get store');
+        } catch (LocalizedException $e) {
+            $this->logger->exception('Could not build category JSON map');
+        }
+
+        return json_encode((object) $array, JSON_UNESCAPED_SLASHES);
+    }
+
+    /**
+     * @param Store $store
+     * @return array
+     * @throws LocalizedException
+     */
+    private function getMagentoCategories(Store $store) {
+
+        $baseUrl = '';
+        $categoriesArray = [];
+
+        try {
+            $baseUrl = $store->getBaseUrl();
```

Comment:
Is this really nulllable?

Reply:
It's not nullable, there's no null check. It throws exception